src/connector/src/connector_common/connection.rs

```diff
@@ -265,6 +273,27 @@ impl Connection for IcebergConnection {
             }
         }
 
+        if self.hosted_catalog.unwrap_or(false) {
+            // If `hosted_catalog` is set, we don't need to test the catalog.
+            // ensure there is not catalog type and catalog uri
+            if self.catalog_type.is_some() {
+                bail!("`catalog.type` must not be set when `hosted_catalog` is set");
+            }
+            if self.catalog_uri.is_some() {
+                bail!("`catalog.uri` must not be set when `hosted_catalog` is set");
+            }
+            if self.catalog_name.is_some() {
+                bail!("`catalog.name` must not be set when `hosted_catalog` is set");
+            }
+            if self.jdbc_user.is_some() {
+                bail!("`catalog.jdbc.user` must not be set when `hosted_catalog` is set");
+            }
+            if self.jdbc_password.is_some() {
+                bail!("`catalog.jdbc.password` must not be set when `hosted_catalog` is set");
+            }
+            return Ok(());
+        }
```

Comment:
> Maybe we should check the config is not set by user here, which will be overrideen by us.

We check it when creating iceberg connection cc @xxchan

Reply:
LGTM 👍